ground/src/main/java/com/google/android/ground/ui/common/BaseMapViewModel.kt

```diff
@@ -72,6 +72,9 @@ constructor(
   @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
 ) : AbstractViewModel() {
 
+  /** This is set to true as soon as the location lock is disabled. */
+  private var isLocationLockReset = true
```

Comment:
In that case should this be named for the positive case, `locationLockEnabled`?

Reply:
The intention is to mark the field as false as soon as we consume it. `locationLockEnabled` would mean that we can't set the bit to false after it's first use.

Reply:
I see. In any case I think the functional approach described below is easier to understand than managing an additional state and relying on side-effects.

Reply:
This is now reverted